File: Codility/BinaryGap.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

int solution(int N) {
    string str;
    while (N != 0) {
        if (N % 2 == 0) str.push_back('0');
        else str.push_back('1');
        N /= 2;
    }
    
    bool isOne = false;
    int maxCnt = 0, cnt = 0;
    for (int i = 0; i < str.size(); i++) {
        if (str[i] == '0' && isOne == false) continue;
        if (str[i] == '1') {
            isOne = true;
            maxCnt = max(maxCnt, cnt);
            cnt = 0;
        }
        else {
            cnt++;
        }
    }
    return maxCnt;
}

int main()
{
    cout << solution(5);
}

File: Java/src/codility/BinaryGap.java
package codility;

public class BinaryGap {

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution(1041));
    }
}

class Solution {
    public int solution(int N) {
        int cur = 0, cnt = 0, res = 0;
        boolean isOne = false;
        while(N!=0){
            cur = N % 2;
            if(cur==1){
                if(!isOne) isOne = true;
                else res = Math.max(res,cnt);
                cnt = 0;
            }
            else cnt++;
            N /= 2;
        }
        return res;
    }
}